Server/middlewares/loginChecker.js

```diff
@@ -0,0 +1,50 @@
+const jwt = require("jsonwebtoken");
+const asyncHandler = require("express-async-handler");
+const User = require('../models/user');
+
+
+/**
+ * Middleware for checking user authentication and decoding JWT tokens.
+ * 
+ * This middleware function that verifies the presence of a valid JWT token
+ * It decodes the token, checks its validity, and attaches
+ * the decoded user information to the request object if successful. If the token is missing or invalid,
+ * it returns an unauthorized request error.
+ */
+
+
+const loginChecker = asyncHandler(async (req, res, next) => {
```

Comment:
Would be more readable if if's were nested less
if ( !( req.headers.authorization...) {
   do else block
   return
if (! token)
   do else block
   return
jwt.verify(...)

Reply:
Done